tests/test_datasets.py

```diff
@@ -23,6 +24,60 @@ def test_constant(func, n_samples, n_features, n_bkps, noise_std):
     assert bkps[-1] == n_samples
 
 
```

Comment:
Use `pytest.mark.parametrize` to have a simpler test.

```suggestion
@pytest.mark.parametrize("signal_generator", [pw_constant, pw_normal, pw_wavy])
```

Reply:
Not all functions have the same signature, so the parametrization might not be as elegant as it seems. I will look at this to fine a better way to do that !

Reply:
True, forget about it then.